rir/src/compiler/pir/type.cpp

```diff
@@ -1,11 +1,17 @@
 #include "type.h"
+#include "../../interpreter/LazyEnvironment.h"
+#include "../parameter.h"
 #include "R/r.h"
 
 extern "C" Rboolean(Rf_isObject)(SEXP s);
 
 namespace rir {
 namespace pir {
 
+bool Parameter::RIR_CHECK_PIR_TYPES =
+    getenv("RIR_CHECK_PIR_TYPES") ? (bool)atoi(getenv("RIR_CHECK_PIR_TYPES"))
+                                  : IS_SLOWASSERT;
```

Comment:
I find this very unexpected. please only enable it when RIR_CHECK_PIR_TYPES is set.